source/opt/loop_descriptor.cpp

```diff
@@ -268,7 +268,7 @@ bool Loop::IsBasicBlockInLoopSlow(const BasicBlock* bb) {
   assert(bb->GetParent() && "The basic block does not belong to a function");
   DominatorAnalysis* dom_analysis =
       context_->GetDominatorAnalysis(bb->GetParent());
-  if (dom_analysis->IsReachable(bb) &&
+  if (context_->IsReachable(*bb) &&
```

Comment:
This should probably still be called from the `dom_analysis`, right?  You end up repeating work if you do it through the context.

Reply:
Agreed - done.